### PR TITLE
Update semantic convention constants generator script and template

### LINKED ISSUE #3042: Update semantic constants generator template
The [template] for generating constants for semantic conventions needs to be updated to work with the latest version of the YAML format and generator tool (https://github.com/open-telemetry/build-tools/releases/tag/v0.3.0) in order to update the spec reference to the next release (after https://github.com/open-telemetry/opentelemetry-specification/pull/1550).

[template]: https://github.com/open-telemetry/opentelemetry-java/blob/main/buildscripts/semantic-convention/templates/SemanticAttributes.java.j2

### PR DESCRIPTION
**Pull request by arminru**
Fixes #3042.

### CODE REVIEW

File: buildscripts/semantic-convention/generate.sh
Comment (arminru) [inline buildscripts/semantic-convention/generate.sh:7]: We could also put this on hold until spec v1.2.0 is released and re-generate the constants again if it includes updates.
```suggestion
SPEC_VERSION=v1.2.0
```

Comment (jkwatson) [reply buildscripts/semantic-convention/generate.sh:7]: yes, we do not want to run against un-released spec versions now that we've hit 1.0. I'll add a tag to this PR that it's blocked on the spec release.

**Review by jkwatson — DISMISSED** [review]
Blocking this until the specification has been released that includes the relevant changes.

File: buildscripts/semantic-convention/generate.sh
Comment (arminru) [reply buildscripts/semantic-convention/generate.sh:7]: There were no semconv changes other than the number->int one (https://github.com/open-telemetry/opentelemetry-specification/compare/v1.1.0...a44d863edcdef63b0adce7b47df001933b7a158a) and thus the *Attributes.java files were not touched when I ran the script. Either is fine with me.

Comment (jkwatson) [reply buildscripts/semantic-convention/generate.sh:7]: I want to draw a pretty hard line of not updating things until the spec is released. We should not try to chase spec updates, and I don't want maintainers to have to verify shas in order to confirm that there aren't any "substantive" changes.

**Comment by jkwatson** [comment]
ok, the spec has now had 1.2.0 released, so this can move forward. @arminru do you want to update it, or should I open a new one with your changes included?

**Comment by arminru** [comment]
@jkwatson I rebased and updated the PR.

**Review by jkwatson — APPROVED** [review]
thanks!

**Pull request merged**
